src/commands/ls.ts

```diff
@@ -48,15 +50,36 @@ export const lsCommand = (yargs: yargs.Argv) =>
     fsShutdownGuard(ls)
   );
 
+/**
+ * If the user has requested a size, replace it with double the requested size,
+ * otherwise request double the default.
+ *
+ * This is done so that we can give the user a sense of the number of results
+ * that are not displayed.
+ */
+const convertLsSizeArg = (args: string[]) => {
+  const convertedArgs = [...args];
+  const sizeIndex = convertedArgs.findIndex((a) => a === "--size");
+  const requestedSize = +(
+    (sizeIndex >= 0
+      ? pullAt(convertedArgs, sizeIndex, sizeIndex + 1)[1]
+      : undefined) ?? DEFAULT_RESPONSE_SIZE
+  );
+  convertedArgs.push("--size", String(requestedSize * 2));
+  return { convertedArgs, requestedSize };
+};
+
 const ls = async (
   args: yargs.ArgumentsCamelCase<{
     arguments: string[];
   }>
 ) => {
   const authn = await authenticate();
+  const { convertedArgs, requestedSize } = convertLsSizeArg(args.arguments);
+
   const data = await spinUntil(
     "Listing accessible resources",
-    fetchCommand<LsResponse>(authn, args, ["ls", ...args.arguments])
+    fetchCommand<LsResponse>(authn, args, ["ls", ...convertedArgs])
```

Comment:
im a bit iffy on this approach, naively i would have expected the `fetch` command to return the total number of results before performing fuzzy search. why are we taking this "doubling" approach? 🤔

Reply:
appreciate the extra context. maybe in the future we can improve our current architecture to support this but im on board with these changes.